### ISSUE TITLE
test.cb.beforeEach and test.cb.afterEach throw error even though `t.end()` is defined

### ISSUE THREAD

**Issue by niftylettuce** [avajs/eslint-plugin-ava#180]
Even though I explicitly call `t.end()` in my `beforeEach` and `afterEach` functions, this error is still thrown:

```sh
Callback test was not ended. Make sure to explicitly end the test with `t.end()`. (ava/test-ended)
```

**Comment by novemberborn** [comment]
Could you share an example?

**Comment by niftylettuce** [comment]
```js
test.cb.beforeEach(t => {
  setTimeout(() => t.end(), 3000);
});
```

**Comment by novemberborn** [comment]
@niftylettuce this works for me:

```js
import test from 'ava'

test.cb.beforeEach(t => {
  setTimeout(() => t.end(), 3000);
});

test(t => t.pass())
```

If you're using `test.cb()` with a test, as well as in the before and after hooks, you still have to call `t.end()` inside the test itself.

**Comment by niftylettuce** [comment]
I know that, but it still didn't work.  Also try putting the test in a
separate file and requiring it.  That's what triggered it for me.

**Comment by novemberborn** [comment]
@niftylettuce could you provide a full example of where this is failing for you?

**Comment by novemberborn** [comment]
@niftylettuce were you referring to the example you posted at https://github.com/avajs/ava/issues/1687#issue-294380067?

This seems to be similar to #119. I think `test-ended` should not error if a function reference is passed. Perhaps it could warn, but honestly that doesn't seem useful either.

**Comment by sindresorhus** [comment]
Closing as `test.cb` is no longer a thing.
